### PR TITLE
fixed tests

### PR DESCRIPTION
**Pull request by TamarAuf**

### CODE REVIEW

File: src/test/java/touro/snake/SnakeTest.java [hunk 1]
```diff
@@ -132,12 +133,13 @@ public void inBounds_false() {
         //given
         SnakeHeadStateMachine snakeHeadStateMachine = mock(SnakeHeadStateMachine.class);
         Snake snake = new Snake(snakeHeadStateMachine);
+        when(snakeHeadStateMachine.getDirection()).thenReturn(mock(Direction.class));
```
Comment (andrewoid) [inline src/test/java/touro/snake/SnakeTest.java:136]: Don't use a mock Direction here, the code needs an actual direction to move or else the Snake can't move.

File: src/test/java/touro/snake/SnakeTest.java [hunk 2]
```diff
@@ -119,7 +119,7 @@ public void eatsSelf_false() {
     @Test
     public void inBounds_true() {
         //given
-        SnakeHeadStateMachine snakeHeadStateMachine = mock(SnakeHeadStateMachine.class);
+        SnakeHeadStateMachine snakeHeadStateMachine = new SnakeHeadStateMachine(Direction.East);
```
Comment (andrewoid) [inline src/test/java/touro/snake/SnakeTest.java:122]: When writing tests, every class should be a `mock` except the class you are testing. Change this back.

Comment (TamarAuf) [reply src/test/java/touro/snake/SnakeTest.java:122]: > Don't use a mock Direction here, the code needs an actual direction to move or else the Snake can't move.

So this doesn't matter?

Comment (andrewoid) [reply src/test/java/touro/snake/SnakeTest.java:122]: Direction objects are essentially constants.

**Pull request merged**